src/dotnet/APIView/APIViewWeb/Managers/CopilotCommentsManager.cs

```diff
@@ -31,163 +31,136 @@ IConfiguration configuration
+                document.BadCode = badCode;
+                document.Embedding = embedding;
             }
 
-            if (comment != null)
+            if (goodCode != null)
             {
-                update = update.Set("Comment", comment);
+                document.GoodCode = goodCode;
             }
 
-            if (guidelineIds != null)
+            if (comment != null)
             {
-                update = update.Set("GuidelineIds", guidelineIds);
+                document.Comment = comment;
             }
 
-            if (badCode != null)
+            if (guidelineIds != null)
             {
-                var embedding = await GetEmbeddingsAsync(badCode);
-                update = update
-                    .Set("BadCode", badCode)
-                    .Set("Embedding", embedding);
+                document.GuidelineIds = SplitGuidelineIds(guidelineIds);
             }
 
-            return await _copilotCommentsRepository.UpdateDocumentAsync(filter, update);
+            await _copilotCommentsRepository.UpdateDocumentAsync(document);
+            return GetDocumentJsonWithoutEmbedding(document);
         }
 
-        public Task DeleteDocumentAsync(string user, string id)
+        public Task DeleteDocumentAsync(string id, string language, string user)
         {
-            var filter = GetIdFilter(id);
-
-            var updateBuilder = Builders<CopilotCommentModel>.Update;
-            var update = updateBuilder
-                .Set("IsDeleted", true)
-                .Set("ModifiedOn", DateTime.UtcNow)
-                .Set("ModifiedBy", user);
+            return _copilotCommentsRepository.DeleteDocumentAsync(id, language, user); 
+        }
 
-            return _copilotCommentsRepository.DeleteDocumentAsync(filter, update); 
+        public async Task<string> GetDocumentAsync(string id, string language)
+        {
+            var document = await _copilotCommentsRepository.GetDocumentAsync(id, language);
+            return GetDocumentJsonWithoutEmbedding(document);
         }
 
-        public async Task<string> GetDocumentAsync(string id)
+        public async Task<string> SearchDocumentsAsync(string language, string badCode, float threshold, int limit)
         {
-            var filter = GetIdFilter(id);
-            var document = await _copilotCommentsRepository.GetDocumentAsync(filter);
+            var embeddings = await GetEmbeddingsAsync(badCode);
+            var documents = await _copilotCommentsRepository.SearchLanguage(language);
 
-            var documentJson = JsonConvert.SerializeObject(document);
+            var searchResults = new List<CopilotSearchModel>();
+            foreach (var document in documents)
+            {
+                var similarity = CosineSimilarity(document.Embedding, embeddings);
+                if (similarity < threshold)
+                    continue;
+                searchResults.Add(new CopilotSearchModel(similarity, document));
+            }
```

Comment:
Mostly just curious, does the cosmos db vector search require you to iterate over each document to check for similarity? Is there no convenience function? With other indexes, I've seen that you just give it the index name and call search with your embedding and it presumably does it for you, returning the top X results

Reply:
edit: scrolled a little and now I see we write our own similarity check. Does cosmos not support a built-in way to do this?

Reply:
Unfortunately yes, as far as I know. The reason we wanted to use Azure Cosmos DB for MongoDB as our database is because they supported a built-in vector search, but they don't support the scoring method (MongoDB Atlas does but we want to stay within Azure), which we need for searching values that are more similar than `threshold`. Which other index were you referencing? I'd love to take a look!

Reply:
Have we tried Azure Cognitive Search? It seems like we can connect our data from Cosmos DB to the search index and then use built-in search queries. https://learn.microsoft.com/en-us/azure/search/search-indexer-overview

Reply:
Thanks I'll give it a shot! I'm not sure if I fully understand search index yet, but I'll give it a try and see if I can get it working.